packages/studio-base/src/players/RosbridgePlayer.ts

```diff
@@ -476,14 +511,133 @@ export default class RosbridgePlayer implements Player {
+          }
+          else if ( message._topic_type === 'nav_msgs/msg/OccupancyGrid' ) {
+            let rdata = message._data_jpeg;
+
+            if ( this.#cachedGrid != undefined )
+                message.data = this.#cachedGrid;
+
+            // Decode the base64 JPEG to pixel data
+            decodeBase64Png(rdata)
+                .then((pixelData) => {
+                    // Assign the decoded RGB pixel data to message.data
+                    const decodedA = new Int8Array(pixelData); // Uint8Array
+
+                    const sumsArray = [];
+                    let alen = decodedA.length;
+                    if (decodedA != undefined) {
+                        for (let i=0; i < alen; i+=3) {
+                            sumsArray.push( 0.3 * (decodedA[i] || 0)
+										  + 0.59 * (decodedA[i+1] || 0)
+										  + 0.11 * (decodedA[i+2] || 0) );
+                        }
+                        this.#cachedGrid = new Int8Array( sumsArray );
+                        message.data = this.#cachedGrid;
+                    }
+
+                })
+                .catch((error) => {
+                    console.error('Error decoding image:', error);
+                });
+          }
+          else if ( message._topic_type === 'sensor_msgs/msg/PointCloud2' ) {
+            let rdata = this._base64decode(message._data_uint16.points)
+            console.log ( rdata );
+            let rview = new DataView(rdata);
+
+            let num_ranges = rdata.byteLength / 6;
+
+            const bounds: number[] = message._data_uint16.bounds;
+            
+            const xmin: number = bounds[0] || 0, xmax: number = bounds[1] || 0;
+            const ymin: number = bounds[2] || 0, ymax: number = bounds[3] || 0;
+            const zmin: number = bounds[4] || 0, zmax: number = bounds[5] || 0;
+            
+            const pointsFloat32: Float32Array = new Float32Array(num_ranges*3);
+
+            for (let i: number = 0; i < num_ranges; i++) {
+              let offset = i * 6;
+              const x: number = rview.getUint16(offset, true);
+              const y: number = rview.getUint16(offset+2, true);
+              const z: number = rview.getUint16(offset+4, true);
+
+              pointsFloat32[i * 3] = this.scaleBackArray(x, xmin, xmax);
+              pointsFloat32[i * 3 + 1] = this.scaleBackArray(y, ymin, ymax);
+              pointsFloat32[i * 3 + 2] = this.scaleBackArray(z, zmin, zmax);
+            }
+
+            //const buffer = paddedPointsFloat32.buffer;
+            const buffer = pointsFloat32.buffer;
+            const points: Uint8Array = new Uint8Array(buffer);
+
+            message.width = pointsFloat32.length/3;
+            message.height = 1;
+            message.point_step = 12;
+            message.row_step = message.width * message.point_step;
+            
+            message.data = points;
+          }
+
+          const innerMessage = message;
+          console.log ( innerMessage );
```

Comment:
split this in different methods, for example, `parse_scan`, `parse_occupancy_grid` and so on so it is easier to read